apiserver/facades/client/highavailability/highavailability_test.go

```diff
@@ -167,6 +187,46 @@ func (s *clientSuite) TestEnableHASeries(c *gc.C) {
 	c.Assert(machines[4].Series(), gc.Equals, "non-default")
 }
 
+func (s *clientSuite) TestEnableHAErrorForNoAddresses(c *gc.C) {
+	machines, err := s.State.AllMachines()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(machines, gc.HasLen, 1)
+	c.Assert(machines[0].Series(), gc.Equals, "quantal")
+
+	enableHAResult, err := s.enableHA(c, 3, emptyCons, defaultSeries, nil)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(enableHAResult.Added, gc.DeepEquals, []string{"machine-1", "machine-2"})
+
+	_, err = s.enableHA(c, 5, emptyCons, defaultSeries, nil)
+	c.Assert(err, gc.ErrorMatches,
+		`juju-ha-space is not set and a unique cloud-local address was not found for machines: \[1 2\]`)
+}
+
+func (s *clientSuite) TestEnableHAErrorForMultiCloudLocal(c *gc.C) {
+	machines, err := s.State.AllMachines()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(machines, gc.HasLen, 1)
+	c.Assert(machines[0].Series(), gc.Equals, "quantal")
+
+	enableHAResult, err := s.enableHA(c, 3, emptyCons, defaultSeries, nil)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(enableHAResult.Added, gc.DeepEquals, []string{"machine-1", "machine-2"})
+
+	s.setMachineAddresses(c, "1")
+
+	m, err := s.State.Machine("2")
+	c.Assert(err, jc.ErrorIsNil)
+	err = m.SetMachineAddresses(
+		network.NewScopedAddress("cloud-local2.internal", network.ScopeCloudLocal),
+		network.NewScopedAddress("cloud-local22.internal", network.ScopeCloudLocal),
+	)
+	c.Assert(err, jc.ErrorIsNil)
+
+	_, err = s.enableHA(c, 5, emptyCons, defaultSeries, nil)
+	c.Assert(err, gc.ErrorMatches,
+		`juju-ha-space is not set and a unique cloud-local address was not found for machines: \[2\]`)
```

Comment:
I'm a little more concerned about the case where machine-0 has multiple addresses and then you go into enable-ha 3 for the first time.
Going from 3 => 5 would be uncommon, and it feels odd to fail at that point.
That said, it does exercise the code path, so I'm not opposed to it. Just that the real-world case is more about no-ha-space being set, and machine-0 has >1 address.

(especially since its almost impossible in real activities to have a machine without *any* address, since we can't do anything with that machine)